Wrappers/Python/cil/framework/acquisition_geometry.py

```diff
@@ -1217,6 +1255,149 @@ def set_centre_of_rotation(self, offset, angle):
         self.rotation_axis.direction = p2_on_plane - p1_on_plane
 
 
+class Cone3D_Flex(SystemConfiguration):
+    r'''This class creates the SystemConfiguration of a cone beam 3D tomographic system
+
+    Parameters
+    ----------
+    source_position_set : array_like
+        N 3D vectors describing the position of the source (x,y,z)
+    detector_position_set : array_like
+        N 3D vectors describing the position of the centre of the detector (x,y,z)
+    detector_direction_x_set : array_like
+        N 3D vectors describing the direction of the detector_x (x,y,z)
+    detector_direction_y_set : array_like
+        N 3D vectors describing the direction of the detector_y (x,y,z)
+    volume_centre_position : array_like, optional
+        A 3D vector describing the position of the centre of the reconstructed volume (x,y,z). Default is [0,0,0].
+    units : str
+        Label the units of distance used for the configuration
+
+    '''
+
+    def __init__ (self, source_position_set, detector_position_set, detector_direction_x_set, detector_direction_y_set, volume_centre_position, units='units'):
+
+        self.num_positions = len(source_position_set)
+        super(Cone3D_Flex, self).__init__(dof=3, geometry = 'cone_flex', units=units, num_vectors= self.num_positions)
+
+        # Check that all input sets have the same length
+        if not (len(detector_position_set) == self.num_positions and
+                len(detector_direction_x_set) == self.num_positions and
+                len(detector_direction_y_set) == self.num_positions):
+            raise ValueError(f"All input sets must have the same length. Got {self.num_positions} source positions, \
+                            {len(detector_position_set)} detector positions, {len(detector_direction_x_set)} detector direction x vectors, \
+                            and {len(detector_direction_y_set)} detector direction y vectors.")
+    
+        # Validate and convert the input arrays
+        for i in range(self.num_positions):
+            self.source[i].position = source_position_set[i]
+            self.detector[i].position = detector_position_set[i]
+            self.detector[i].set_direction(detector_direction_x_set[i], detector_direction_y_set[i])
+
+        self.volume_centre.position = volume_centre_position
+
+
+    def system_description(self):
+        """
+        Returns the system configuration type
+        """
+        return SystemConfiguration.SYSTEM_NONSTANDARD
+
+    def get_centre_slice(self):
+
+        """Returns the 2D system configuration corresponding to the centre slice
+        """
+        raise NotImplementedError("This method is not implemented for Cone3D_Flex")
```

Comment:
I think this a `TypeError` as we are not going to be implementing this code at all as it doesn't make any sense to.